src/components/SingleBlock.tsx

```diff
@@ -49,27 +48,39 @@ function SingleBlock({ constants }: Props) {
 
     const { hash, prev_hash, nonce, total_kernel_offset, version, timestamp } = blockHeader;
     const { accumulated_monero_difficulty, accumulated_blake_difficulty } = blockPow;
-    const { inputs = [], kernels = [], outputs =[] } = blockBody;
+    const { inputs = [], kernels = [], outputs = [] } = blockBody;
 
-   inputs.forEach(i => {
-        i.group = 'inputs';
-        i.color = '#F97C0C';
-        i.size = inputs.length;
-        singleBlockDataArray.push(i);
+    inputs.map((i: InputsEntity) => {
```

Comment:
Why is this a map? There is nothing being set?

Reply:
It's setting the group, color and size:

```
const temp: Inputs = {
            group: 'inputs',
            size: inputs.length,
            color: '#F97C0C',
            ...i
        };
```

Reply:
Either use a forEach and don't return anything or use something like.
```ts
singleBlockDataArray = singleBlockDataArray.concat(inputs.map(i => {
   i.size = inputs.length
  /// etc etc
  return i
}));
```